File: AtCoderBeginnersSelection/ShiftOnly.go
package main

import "fmt"

func main(){
	var N,temp int
	fmt.Scanf("%d",&N)
	var a []int
	for i:=0;i<N;i++ {
		fmt.Scan(&temp)
		a = append(a,temp)
	}
	count := -1
	flag := true
	for flag {
		for i,v := range a {
			if v%2 == 1{
				flag = false
			}
			a[i] /= 2
		}
		count++
	}

	fmt.Println(count)
}

File: AtCoderBeginnersSelection/ShiftOnly.cpp
#include <iostream>
using namespace std;

int main(void){
    int N;
    cin >> N;
    int* a = new int[N];
    for(int i=0;i<N;i++){
        cin >> a[i];
    }
    int count = 0;
    bool flag = true;
    while(flag){
        for(int i=0;i<N;i++){
            if(a[i]%2==0){
                a[i]/=2;
            }else{
                flag = false;
            }
        }
        if(flag)
            count++;
    }
    cout << count << endl;
    delete a;
}